src/OrchardCore.Modules/OrchardCore.Tenants/Views/Admin/Index.cshtml

```diff
@@ -37,7 +46,16 @@
     {
         var shellSettings = shellSettingsEntry.ShellSettings;
         var host = shellSettings.RequestUrlHost?.Split(new[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries).FirstOrDefault();       
-        var hostString = host != null ? new HostString(host) : Context.Request.Host;
+
+        var hostString = Context.Request.Host;
```

Comment:
I'm not sure why we use function here to remove the duplicate code?!!

Reply:
Agreed, reuse code in private method

Reply:
> I'm not sure why we use function here to remove the duplicate code?!!

Yes, in this razor view we have 2 `@function`s with specific code, one is called twice, so yes to not have duplicate code, and one is called once but to still not have too much specific code mixed with html tags.

> Agreed, reuse code in private method

Not sure I undestand, we already reuse this code through this razor `@function` being a kind of private method.

Reply:
Take this part of the code and create a 3rd function `BuildUrlWithHost`